### PR TITLE
#31: Improve stability when deleting files

### LINKED ISSUE #31: Deleting files fails with SlowDown error
Deleting files sometimes fails with the following error message:
```
 S3DocumentFilesAdapterIT.testManySmallJsonFiles:153->createTestSetupWithSmallJsonFiles:228 » IllegalState Deleting files from bucket persistent-small-json-files-test-fixture failed: {"errorType":"SlowDown","errorMessage":"Please reduce your request rate.","trace":["SlowDown: Please reduce your request rate.","    at throwDefaultError (/var/runtime/node_modules/@aws-sdk/smithy-client/dist-cjs/default-error-handler.js:8:22)","    at /var/runtime/node_modules/@aws-sdk/smithy-client/dist-cjs/default-error-handler.js:18:39","    at de_ListObjectsV2CommandError (/var/runtime/node_modules/@aws-sdk/client-s3/dist-cjs/protocols/Aws_restXml.js:5105:20)","    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","    at async /var/runtime/node_modules/@aws-sdk/middleware-serde/dist-cjs/deserializerMiddleware.js:7:24","    at async /var/runtime/node_modules/@aws-sdk/middleware-signing/dist-cjs/awsAuthMiddleware.js:14:20","    at async /var/runtime/node_modules/@aws-sdk/middleware-retry/dist-cjs/retryMiddleware.js:27:46","    at async /var/runtime/node_modules/@aws-sdk/middleware-logger/dist-cjs/loggerMiddleware.js:7:26","    at async listFiles (file:///var/task/createJsonFilesLambda.js:222:20)","    at async handleDelete (file:///var/task/createJsonFilesLambda.js:152:24)"]}
```

See log https://github.com/exasol/s3-document-files-virtual-schema/actions/runs/7210829196/job/19644842441

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #31

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -121,23 +121,58 @@
...
+                <artifactId>maven-clean-plugin</artifactId>
+                <version>3.3.2</version>
                 <configuration>
-                    <excludeVulnerabilityIds>
-                        <!-- Transitive dependency io.netty:netty-handler of software.amazon.awssdk.
-                            According to https://github.com/aws/aws-sdk-java-v2/issues/4584#issuecomment-1760547020
-                            we are not affected. -->
-                        <exclude>CVE-2023-4586</exclude>
-                    </excludeVulnerabilityIds>
+                    <filesets>
```
Comment (ckunki) [inline pom.xml:127]: Here I would need some explanation.

File: src/main/resources/createJsonFilesLambda/createJsonFilesLambda.js
Comment (ckunki) [inline src/main/resources/createJsonFilesLambda/createJsonFilesLambda.js:1]: Also for this file I would appreciate a short explanation / introduction.

**Review by ckunki — CHANGES_REQUESTED** [review]

File: src/main/resources/createJsonFilesLambda/createJsonFilesLambda.js [hunk 2]
```diff
@@ -123,15 +131,19 @@ function reverseString(str) {
...
  */
 async function uploadFile(bucket, key, content) {
     const command = new PutObjectCommand({
         Bucket: bucket,
         Key: key,
         Body: content
     });
-    return doWithRetry(`Upload file ${key}`, () => s3.send(command));
+
+    return doWithRetry(`Upload file ${key}`, async function upload() {
```
Comment (ckunki) [inline src/main/resources/createJsonFilesLambda/createJsonFilesLambda.js:143]: Could this be an anonymous function?

**Review by ckunki — APPROVED** [review]

File: src/main/resources/createJsonFilesLambda/createJsonFilesLambda.js (hunk 2 above)
Comment (kaklakariada) [reply src/main/resources/createJsonFilesLambda/createJsonFilesLambda.js:143]: Sorry, won't fix

**Pull request merged**